### PR TITLE
ProgressTableDetailView unit tests

### PR DESCRIPTION
**Pull request by maureensturgeon**
- Add unit tests for `ProgressTableDetailView`
- Add progress data test helpers
- Refactor `ProgressTableDetailCellTests` to use test helpers to generate data

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailViewTest.jsx
Comment (cforkish) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailViewTest.jsx]: NIT: typo

**Review by cforkish — APPROVED** [review]
these tests look great. i'm wondering about the test data you're using here though. it looks like data that you might want to reuse in other tests, in which case i'd suggest moving it to `sectionProgressTestHelpers`.

the more hand-coded test data we have, the more things we need to update when we change our data shapes (something i've been dealing with _extensively_ in this refactor work). so i'd like us to reuse as much test data as possible between our tests.

i'm going ahead and approving this in case this data is actually specific to this test. but if it is data that could be reused, please update accordingly before merging.

**Comment by maureensturgeon** [comment]
> these tests look great. i'm wondering about the test data you're using here though. it looks like data that you might want to reuse in other tests, in which case i'd suggest moving it to `sectionProgressTestHelpers`.
> 
> the more hand-coded test data we have, the more things we need to update when we change our data shapes (something i've been dealing with _extensively_ in this refactor work). so i'd like us to reuse as much test data as possible between our tests.
> 
> i'm going ahead and approving this in case this data is actually specific to this test. but if it is data that could be reused, please update accordingly before merging.

Yes! I totally agree - I think I may do this once I have more tests merged in so that I can figure out what's useful to abstract, this has been on my mind as I've been tackling the larger components that need extensive data sets.

**Comment by cforkish** [comment]
okay. let's wait to merge this until you have that figure out then. i want “tests merged into 1604” to mean we’re ready to open a PR for that component into staging, but to do that we need to know what helper files to include in the PR. i will focus on PRs for the header components in the meantime. nbd if some of the test fixtures we merge into staging get updated by future PRs, but still better to hold off on these higher level components i think.

**Review by cforkish — APPROVED** [review]
LGTM!

**Pull request merged**
